text/2022-02-14-link-application-metrics-with-chaos-experiments.md

```diff
@@ -0,0 +1,249 @@
+panel would be copied to the new panel.
+
+> The feature of duplicate panel is helpful to create series of panels with same
+> Grafana Panel with different variables
+
+#### Remove a Panel
+
+User could remove the panel imported from Grafana. When the panel is removed, it
+would also be removed from the linked chaos experiment.
+
+### Provision Grafana Panel then creating chaos experiment
+
+As so far, user could only link Grafana Panel to already existing chaos
+experiments. Here is a way to create the link of Grafana Panel to a new chaos
+experiment as the chaos experiment is created.
+
+We would introduce a new annotation: `metrics.chaos-mesh.org/link-panel`, the
+value of this annotation should be an array of names of Grafana Panels as JSON
+string. Once one chaos experiment holds this annotation, links between
+corresponding Grafana Panels and this chaos experiment would be created.
+
+The behavior of this annotation is "appending only", it means if user remove the
+annotation of the chaos experiment, the links would still eixst.
+
+### About Authorization and Security
+
+Importing panel from Grafana requires a Grafana API Key with role `Viewer`.
+
+Chaos Mesh does not contains any user management system, so all the Grafana
+backend and imported panels are simply shared with all users.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+We already have a grafana plugin as mark annotations on grafana dashboard. There
+are also feature overlapping between that grafana plugin and this new feature.
+It might take more effort to keep behaviors consistent.
+
+Integrating Grafana into our components seems not really simple, most of
+powerful features of Grafana, like variables and templating and drawing the
+charts, are implemented in frontend. And Grafana did not well export them as a
+library.
+
+## Alternatives
+
+<!-- - Why is this design the best in the space of possible designs?
+- What other designs have been considered and what is the rationale for not
+  choosing them?
+- What is the impact of not doing this? -->
+
+- Import metrics data from datasource directly instead of proxy by Grafana, then
+  draw charts by ourself. The most important problem is that we also provide the
+  feature about "exploring the query" and "design panels and dashboards" like
+  Grafana, for user to tuning their charts.
+- Choose other Grafana alternative integration, like Datadog, Elastic stack, New
+  Relic or other. I think we would import panel from more platform other than
+  Grafana, especially if we want to connect with cloud provider and other Cloud
+  Monitoring Services. But for now, using Grafana just makes more sense.
+
+### Alternative: use ConfigMap instead of database
+
+From inspired of "Status Check Template", the resource of "Grafana Panel" could
+also be stored as ConfigMaps.
+
+There are some Pros and Cons.
+
+Pros:
+
+- It would prevent data loss when the Pod chaos-dashboard is deleted/recreated.
```

Comment:
Maybe it's time to reconsider how to treat our data in chaos-dashboard?

Was it designed as persisted or volatile? @g1eny0ung